docs/howtos/extending/magicgui.md

```diff
@@ -146,25 +156,80 @@ viewer.window._qt_window.resize(1225, 900)
 nbscreenshot(viewer, alt_text="A magicgui threshold widget")
 ```
 
-Below we first document how to use napari-specific
-[parameter](magicgui-parameter-annotations) and
-[return type](magicgui-return-annotations) annotations to easily create your own
-widgets. We then explain how to use
-[`magicgui` function widgets in plugin widget contributions](magicgui-plugin-widgets).
-
 ```{note}
 For a more complex example of a {func}`magicgui.magicgui` widget, see the
 [gaussian blur example](https://pyapp-kit.github.io/magicgui/generated_examples/napari/napari_parameter_sweep/#napari-parameter-sweeps)
 in the `magicgui` documentation.
 ```
 
+See [](magicgui-parameter-annotations) for details on how to use `napari` types
+to get information from the napari viewer into your widget. See
+[](magicgui-return-annotations) for information how to use `napari` types to add
+output to the napari viewer.
+
+Note all type annotations *require* that the resulting widget be added to a
```

Comment:
What does this mean? Why does the widget have to be added to a viewer?

Reply:
This was definitely from Talley, and I wonder if it is a `magicgui` thing? Where the type annotation magic doesn't work until its added to a window? :shrug:

Reply:
Amended, hopefully that is okay